unsupported/Eigen/CXX11/src/Tensor/TensorConvolution.h

```diff
@@ -854,9 +866,9 @@ struct TensorEvaluator<const TensorConvolutionOp<Indices, InputArgType, KernelAr
     typedef typename TensorEvaluator<InputArgType, GpuDevice>::Dimensions InputDims;
 
     const int maxSharedMem = m_device.sharedMemPerBlock();
-    const int maxThreadsPerBlock = m_device.maxCudaThreadsPerBlock();
-    const int maxBlocksPerProcessor = m_device.maxCudaThreadsPerMultiProcessor() / maxThreadsPerBlock;
-    const int numMultiProcessors = m_device.getNumCudaMultiProcessors();
+    const int maxThreadsPerBlock = m_device.maxGpuThreadsPerBlock();
+    const int maxBlocksPerProcessor = m_device.maxGpuThreadsPerMultiProcessor() / maxThreadsPerBlock;
+    const int numMultiProcessors = m_device.getNumGpuMultiProcessors();
     const int warpSize = 32;
```

Comment:
wouldn't we need to use `warpSize = 64` on ROCm?

Reply:
It can require more kernel changes if we change the default warpSize, should be cautious.

Reply:
I think we should use `warpSize = 64` on ROCm, but do not undderstand this code good enough to make the change. This struct/routine uses the `warpSize` variable in someplaces and uses a hardocoded `32` / `31` in others. It does not seem that the distinction between using the variable or the hardcoded numbers is by design :(

I will leave this as 32 for ROCm for now...maybe change it later when I understand the code better